pyfr/libs/conduit.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from ctypes import (POINTER, c_char_p, c_void_p, 
+                    c_double, c_float, c_long, c_longlong, RTLD_GLOBAL)
+from pyfr.ctypesutil import LibWrapper
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+    _mode = RTLD_GLOBAL
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (None, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_char_p, 'conduit_node_fetch_path_as_char8_str', c_void_p, 
+         c_char_p),
+        (POINTER(c_float), 'conduit_node_fetch_path_as_float32_ptr', c_void_p,
+         c_char_p),
+        (POINTER(c_double), 'conduit_node_fetch_path_as_float64_ptr', 
+         c_void_p, c_char_p),
+        (POINTER(c_longlong), 'conduit_node_fetch_path_as_int64_ptr', 
+         c_void_p, c_char_p),
+        (c_longlong, 'conduit_node_total_bytes_allocated', c_void_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64', c_void_p, c_char_p,
+         c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         POINTER(c_longlong), c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p), # maybe this is copying, external might be better
+        (c_void_p, 'conduit_node_print', c_void_p),
+        (c_void_p, 'conduit_node_print_detailed', c_void_p),
+    ]
+
+
+class Conduit:
+    def __init__(self):
+        self.lib = ConduitWrappers()
+
+
+class ConduitNode:
+    def __init__(self, ptr=None):
+        cd = Conduit()
+        self.lib = cd.lib
+
+        if ptr is not None:
+            self.ptr = ptr
+        else:
+            self.ptr = c_void_p(self.lib.conduit_node_create(None))
+
+        self._set_ptr = {ConduitNode: self.lib.conduit_node_set_path_node,
```

Comment:
In-line these into the if/else block as they are only used in one place.  (If you like you can also try the new match/case feature in Python 3.10 which greatly simplifies these if/elif/else blocks.)

Reply:
Sure, but the numpy cases will take up more space.